dnf/yum/rpmtrans.py

```diff
@@ -241,12 +253,12 @@ def _extract_cbkey(self, cbkey):
         for tsi in self.base.transaction:
             # only walk the tsis once. prefer finding an erase over an obsoleted
             # package:
-            if tsi.erased is not None and str(tsi.erased) == te.NEVRA():
+            if tsi.erased is not None and str(tsi.erased) == self._te_nevra(te):
                 return tsi.erased, tsi._erased_history_state, tsi
-            if tsi.installed is not None and str(tsi.installed) == te.NEVRA():
+            if tsi.installed is not None and str(tsi.installed) == self._te_nevra(te):
                 return tsi.installed, tsi._installed_history_state, tsi
```

Comment:
I guess that 
```
if tsi.installed is not None and str(tsi.installed) == self._te_nevra(te):
                 return tsi.installed, tsi._installed_history_state, tsi
```
statement can be removed. It was added by me in 769ffaba6ce70322ebba03ad433ebfc12b85ddc0